Library/QuickPulseSender.ts

```diff
@@ -0,0 +1,51 @@
+import https = require("https");
+import Config = require("./Config");
+import AutoCollectHttpDependencies = require("../AutoCollection/HttpDependencies");
+
+// Types
+import * as http from "http";
+import * as Contracts from "../Declarations/Contracts";
+
+class QuickPulseSender {
+    private _config: Config;
+
+    constructor(config: Config) {
+        this._config = config;
+    }
+
+    public ping(envelope: Contracts.EnvelopeQuickPulse, done: (shouldPOST: boolean) => void): void {
+        this._submitData(envelope, done, "ping");
+    }
+
+    public post(envelope: Contracts.EnvelopeQuickPulse, done: (shouldPOST: boolean, res: http.IncomingMessage) => void): void {
+
+        // Important: When POSTing data, envelope must be an array
+        this._submitData([envelope], done, "post");
+    }
+
+    private _submitData(envelope: Contracts.EnvelopeQuickPulse | Contracts.EnvelopeQuickPulse[], done: (shouldPOST: boolean, res: http.IncomingMessage) => void, postOrPing: "post" | "ping"): void {
+        const payload = JSON.stringify(envelope);
+        var options = {
+            [AutoCollectHttpDependencies.disableCollectionRequestOption]: true,
+            host: "rt.services.visualstudio.com",
```

Comment:
This should probably come from config

Reply:
Re-opening because this still isn't from the passed in Config object.

Reply:
Moved to configuration option